src/main/java/com/julienviet/pgclient/Numeric.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2017 Julien Viet
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.julienviet.pgclient;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Objects;
+
+/**
+ * The Postgres <i>NUMERIC</i> type.
+ */
+public final class Numeric extends Number {
+
+  /**
+   * Constant for the {@code NaN} value.
+   */
+  public static final Numeric NaN = new Numeric(null);
+
+  private BigDecimal value;
+
+  /**
+   * Return a {@code Numeric} instance for the given {@code number}.
+   *
+   * @param number the number
+   * @return the {@code Numeric} value
+   */
+  public static Numeric create(Number number) {
```

Comment:
How would this deal with `number` being null?

Reply:
it should throw NullPointerException but it hasn't been bested